docs/book.md

```diff
@@ -0,0 +1,299 @@
+                                                # ensuring all submitted data to the endpoint validates the schema
+                                                # before getting to your `Handler` function.
+
+          - id                                  # These names must match the `properties` names exactly
+          - user_addresses
+
+        properties:                             # `object`s are permitted to have `properties`. These are translated 
+                                                # into fields in the generated case classes.
+
+          id:                                   # Case matters for `properties`! A heuristic determines whether it's
+                                                # possible to translate a property name into a unique, non-reserved
+                                                # camelCase identifier.
+
+            type: string                        # One of the primitive types supported in the OpenAPI specification.
+                                                # FIXME: Include a link to the specification for types and formats
+
+          user_addresses:                       # Similar to `id`, though `user_addresses` can be safely transformed into
+                                                # `userAddress`, so this is done to expose idiomatic Scala. The underlying
+                                                # marshallers and unmarshallers maintain this mapping for you though,
+                                                # so no chance of protocol violations.
+
+            $ref: '#/definitions/UserAddresses' # Ensures that tye type of `userAddress` will be `IndexedSeq[UserAddress]`
+
+      UserAddresses:
+        type: array
+
+        items:                                  # `items` is a special key for `type: array`, identifying the structure of the
+                                                # sequence members
+
+          $ref: '#/definitions/UserAddress'     # Primitive types could be listed here, but as we're referring to another class,
+                                                # we need to explicitly use a `$ref`. This may change in the future,
+                                                # see https://github.com/twilio/guardrail/issues/76
+
+      UserAddress:
+        type: object
+        properties:
+          line1:
+            type: string
+          line2:
+            type: string
+          line3:
+            type: string
+
+Generating a Server
+===================
+
+Guardrail-generated servers come in two parts: a `Resource` and a `Handler`. The `Resource` contains all the routing logic, and accepts a `Handler` as an argument to the `route` function in order to provide an HTTP service in whatever HTTP framework you're hosting your service in. The following is an example from the akka-http server generator:
+
+    trait UserHandler {
```

Comment:
If you intend to publish this as github markdown, it would probably be useful to mark this section explicitly as scala code so you get syntax highlighting.